core-java-modules/core-java-collections-5/src/main/java/com/baeldung/immutables/ImmutableCollectionsExample.java

```diff
@@ -0,0 +1,35 @@
+package com.baeldung.immutables;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class ImmutableCollectionsExample {
+
+    public Map<String, String> createUnmodifiableMap(Map<String, String> map) {
+
+        Map<String, String> uMap = Collections.unmodifiableMap(map);
+        return uMap;
+    }
+
+    public List<String> createUnmodifiableList(List<String> list) {
+
+        List<String> uList = Collections.unmodifiableList(list);
+        return uList;
+    }
+
+    public Map<String, String> createImmuteableMap() {
+        Map<String, String> map = Map.of("color1", "black", "color2", "red", "color3", "orange");
+        return map;
+    }
+
+    public List<String> createImmuteableList() {
+        List<String> list = List.of("One", "Two", "Three");
+        return list;
+    }
+
+    public List<String> createImmuteableList(List<String> l) {
+        List<String> list = List.copyOf(l);
```

Comment:
I've not had a look at the error for the integration tests but this is a candidate since this method is available from Java 10 onwards.

Reply:
Ok